scipy/stats/_continuous_distns.py

```diff
@@ -8806,6 +8806,135 @@ def _entropy(self, kappa):
+            If True, the integral is corrected by the conditional probability
+            of the integration interval.  The return value is the expectation
+            of the function, conditional on being in the given interval.
+            Default is False.
+
+        Additional keyword arguments are passed to the integration routine.
+
+        Returns
+        -------
+        expect : float
+            The calculated expected value.
+
+        Notes
+        -----
+        The integration behavior of this function is inherited from
+        `scipy.integrate.quad`. Neither this function nor
+        `scipy.integrate.quad` can verify whether the integral exists or is
+        finite. For example ``cauchy(0).mean()`` returns ``np.nan`` and
+        ``cauchy(0).expect()`` returns ``0.0``.  It is often the case in
+        directional statistics to supply complex expontial functions
+        (e.g. ``np.exp(1j*x)``). In such cases, this method returns a
+        complex number.
+
+        Likewise, the accuracy of results is not verified by the function.
+        `scipy.integrate.quad` is typically reliable for integrals that are
+        numerically favorable, but it is not guaranteed to converge
+        to a correct value for all possible intervals and integrands. This
+        function is provided for convenience; for critical applications,
+        check results against other integration methods.
+
+        The function is not vectorized.
+
+        Examples
+        --------
+
+        To understand the effect of the bounds of integration consider
+
+        >>> from scipy.stats import vonmises
+        >>> vonmises(kappa=11).expect(lambda x: 1, lb=0.0, ub=2.0).real
+        0.43424095588998324
+
+        This is close to
+
+        >>> vonmises(kappa=1).cdf(2.0) - vonmises(kappa=1).cdf(0.0)
+        0.4342409558899928
+
+        If ``conditional=True``
+
+        >>> vonmises(kappa=1).expect(lambda x: 1, lb=0.0, ub=2.0, conditional=True).real
+        0.999999999999978
+
+        The slight deviation from 1 is due to numerical integration.
+
+        The mean direction can be found as
+
+        >>> r = vonmises(kappa=1).expect(lambda x: np.exp(1j*x))
+        >>> np.angle(r)
+        0.0
+
+        """
+        lockwds = {'loc': loc,
+                   'scale': scale}
+        self._argcheck(*args)
+        _a, _b = -np.pi, np.pi
+        if func is None:
+            def fun(x, *args):
+                return np.exp(1j*x) * self.pdf(x, *args, **lockwds)
+        else:
+            def fun(x, *args):
+                return func(x) * self.pdf(x, *args, **lockwds)
+        def refun(x, *args):
+            return np.real(fun(x, *args))
+        def imfun(x, *args):
+            return np.imag(fun(x, *args))
+        if lb is None:
+            lb = loc + _a
+        if ub is None:
+            ub = loc + _b
+        if ub-lb > 2*np.pi:
+            raise ValueError("Interval of Von Mises integral must be 2pi.")
```

Comment:
interval length must be 2pi or smaller

Reply:
I addressed this in subsequent commits on this PR.